### PR TITLE
chore(config): mark `scriptDataOpts` field as deprecated

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This marks the `extras.scriptDataOpts` field on the Stencil config as deprecated and sprinkles TODO here and there to note where we'll need to rip the related code out later.

BREAKING CHANGES: The `extras.scriptDataOpts` field is deprecated and will be removed entirely in a future version. For now, it is renamed to `__deprecated__scriptDataOpts`.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

This marks the `scriptDataOpts` field as deprecated. hooray for deprecation!

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by tanner-reits — APPROVED** [review]

**Comment by rwaskiewicz** [comment]
@alicewriteswrongs I'm not sure I understand how this relates back to dropping support for legacy browser support. I'm not super familiar with this flag beyond our public facing documentation - can you expound a little on why we need to remove this flag as a part of the larger initiative to deprecate and later drop support for browsers like IE11?

**Comment by alicewriteswrongs** [comment]
mmm yeah it probably doesn't...just had that in as boilerplate from the other PRs.

Should we just say 'its deprecated' and leave it at that?

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -16,8 +16,9 @@ This is a comprehensive list of the breaking changes introduced in the major ver
     * [`dist-custom-elements` Type Declarations](#dist-custom-elements-type-declarations)
   * [Legacy Browser Support Fields Deprecated](#legacy-browser-support-fields-deprecated)
     * [`dynamicImportShim`](#dynamicimportshim)
-    * [`cssVarShim`](#cssvarshim)
+    * [`cssVarsShim`](#cssvarsshim)
```
Comment (alicewriteswrongs) [inline BREAKING_CHANGES.md:19]: noticed a little typo here, oops

**Comment by alicewriteswrongs** [comment]
I think we need to further evaluate whether it makes sense to deprecate this field before we can do so, so I'm going to close this for now.

**Pull request closed without merging**